src/core/loss_detection.c

```diff
@@ -595,25 +596,28 @@ QuicLossDetectionOnPacketAcknowledged(
             PacketSizeFromUdpPayloadSize(
                 QuicAddrGetFamily(&Path->RemoteAddress),
                 Packet->PacketLength);
-
+        BOOLEAN ChangedMtu = FALSE;
         if (!Path->IsMinMtuValidated &&
-            Packet->PacketLength >= QUIC_INITIAL_PACKET_LENGTH) {
+            PacketMtu >= Path->Mtu) {
             Path->IsMinMtuValidated = TRUE;
+            ChangedMtu = TRUE;
             QuicTraceLogConnInfo(
                 PathMinMtuValidated,
                 Connection,
                 "Path[%hhu] Minimum MTU validated",
                 Path->ID);
         }
 
-        if (PacketMtu > Path->Mtu) {
-            Path->Mtu = PacketMtu;
-            QuicTraceLogConnInfo(
-                PathMtuUpdated,
-                Connection,
-                "Path[%hhu] MTU updated to %hu bytes",
-                Path->ID,
-                Path->Mtu);
+        if (Packet->Flags.IsMtuProbe) {
+            CXPLAT_DBG_ASSERT(Path->IsMinMtuValidated);
+            if (QuicMtuDiscoveryOnAckedPacket(
+                    &Path->MtuDiscovery,
+                    PacketMtu,
+                    Connection)) {
+                ChangedMtu = TRUE;
+            }
+        }
+        if (ChangedMtu) {
```

Comment:
Perhaps, we just do this if `Path->IsActive` too?

Reply:
Internally, QuicDatagramOnSendStateChanged always computes off of the active path, and only actually fires an event if things update. So this should be fine for now.

Reply:
Then there isn't any reason to call the function below if it's not for active, right?

Reply:
The old code called it for all paths that changed, not just the active one. I worry if we get to that point, and add multipath, this is a place we might miss.